src/v2i-hub/SpatPlugin/src/SpatPlugin.cpp

```diff
@@ -164,14 +164,23 @@ namespace SpatPlugin {
 		auto spatPtr = (SPAT*)calloc(1, sizeof(SPAT));
 		tmx::messages::SpatEncodedMessage spatEncoded;
 		scConnection->receiveBinarySPAT(spatPtr, PluginClientClockAware::getClock()->nowInMilliseconds());
-		// SpatMessage and SpatEncodeMsg assume responsibilty for SPAT pointers (see constructor documentation for TmxJ2735Message and TmxJ2735EncodedMessage)
-		tmx::messages::SpatMessage spatMessage(spatPtr);
-		spatEncoded.initialize(spatMessage);
+		// SpatMessage assume responsibilty for SPAT pointers (see constructor documentation for TmxJ2735Message)
+		tmx::messages::SpatMessage _spatMessage(spatPtr);
+		MessageFrameMessage frame(_spatMessage.get_j2735_data());
+		spatEncoded.set_data(TmxJ2735EncodedMessage<SPAT>::encode_j2735_message<codec::uper<MessageFrameMessage>>(frame));
 		spatEncoded.addDsrcMetadata(tmx::messages::api::msgPSID::signalPhaseAndTimingMessage_PSID);
 		spatEncoded.set_flags(IvpMsgFlags_RouteDSRC);
 		auto rMsg = dynamic_cast<routeable_message*>(&spatEncoded);
 		BroadcastMessage(*rMsg);
-
+	
+		// TODO Fix j2735::j2735_cast used in TmxJ2735Message(const std::shared_ptr<OtherMsgType> &other)
+		// constructor which allocates memory for wrapping Message Frame (see J2735MessageTemplate.hpp)
+		// without a mechanism to  free it later causing memory leak here if we attempt to initialize a 
+		// EncodeSPatMessage here with the SpatMesssage directly. Adding a deleter to _j2735_data
+		// shared_ptr seems to solve the issue here but causes issues for other unit tests and potentially 
+		// else where. Due to this known issue we need to create MessageFrameMessage manualy and ensure pointer 
+		// to underlying MessageFrame is freed explicitly to avoid memory leak.
+		free(frame.get_j2735_data().get());
```

Comment:
Can we check this pointer exist before we  free them?

Reply:
It does exist. We create it above. I do not think this is necessary. Additionally even if an earlier call freed this pointer, checking with an if condition
```cpp
if (frame.get_j2735_data().get())
```
would still return true unless we set the pointer to nullptr after freeing it.

For this reason I think a check on the pointer is pointless since we know it exists (we create it in this function) and even it it was freed somewhere else, this check would still return true